aws_advanced_python_wrapper/utils/rdsutils.py

```diff
@@ -210,17 +227,38 @@ def identify_rds_type(self, host: Optional[str]) -> RdsUrlType:
 
         return RdsUrlType.OTHER
 
-    def _contains(self, host: str, patterns: list) -> bool:
-        if not host or not host.strip():
-            return False
-
-        return len([pattern for pattern in patterns if search(pattern, host)]) > 0
-
     def _find(self, host: str, patterns: list):
         if not host or not host.strip():
             return None
 
         for pattern in patterns:
+            match = RdsUtils.CACHE_DNS_PATTERNS.get(host)
+            if match:
+                return match
+
             match = search(pattern, host)
             if match:
+                RdsUtils.CACHE_DNS_PATTERNS[host] = match
                 return match
+
+        return None
+
+    def _get_regex_group(self, pattern: Match[str], group_name: str):
+        return pattern.group(group_name)
+
+    def _get_group(self, host: str, group: str):
+        if not host or not host.strip():
+            return None
+
+        pattern = self._find(host, [RdsUtils.AURORA_DNS_PATTERN,
```

Comment:
I think _find could return None, do we need to check for None before passing to get_regex_group?